### PR TITLE
CURA-3540 update profiles when deserializing them

### PR DESCRIPTION
**Pull request by LipuFei**
Create PR for now to facilitate reviewing and discussion.

What it does:
- Before deserialize a profile, we try to update it first
- Did some hacking to ensure as little change as possible
- I added some **FIXME** marks on the things I'm sure if they are correct.

### CODE REVIEW

File: UM/Settings/ContainerStack.py [hunk 1]
```diff
@@ -244,17 +244,37 @@ def serialize(self):
...
 
+    def _readAndValidateSerialized(self, serialized: str) -> configparser.ConfigParser:
+        parser = configparser.ConfigParser(interpolation=None, empty_lines_in_values=False)
+        parser.read_string(serialized)
+
+        if "general" not in parser or any(pn not in parser["general"] for pn in ("version", "name", "id")):
+            raise InvalidContainerStackError("Missing required section 'general' or 'version' property")
+
+        return parser
+
```
Comment (LipuFei) [inline UM/Settings/ContainerStack.py]: Documented in the last commit, right above the `def` line.

**Comment by awhiemstra** [comment]
Note that this causes almost all the Settings unit tests to fail, since you changed the main interface. This should not be merged before those are fixed.

**Comment by awhiemstra** [comment]
Uhm, what happened to "do not merge this until the unit tests are fixed"?

**Comment by nallath** [comment]
You didn't put it back into to-do? I missed this comment and the code worked.

**Comment by LipuFei** [comment]
@nallath @awhiemstra I've fixed the unit tests. This is the commit: https://github.com/Ultimaker/Uranium/commit/de107ee5b374ded641dbc644f28e7bd8276fae72

**Pull request merged**